vtysh/vtysh_main.c

```diff
@@ -417,9 +417,9 @@ int main(int argc, char **argv, char **env)
 			"NOT SUPPORTED since its\nresults are inconsistent!\n");
 	}
 
-	snprintf(vtysh_config, sizeof(vtysh_config), "%s%s%s", sysconfdir,
```

Comment:
This was intentionally removed in e82314b1bf75ee7240e70a8e97aac8dccbe5d061 why do we need to put this back?

Reply:
Ok, interesting. The issue is that by specifying the sysconfdir ( `--config_dir` ), it's missing a `/`.  (the user specified string in the `--config_dir` option has normally no `/` at the end)

It seems the commit e82314b fixed the other issue, but broke the `--config_dir` option.

Snap packages need this option to save to correct location (which is only known at runtime)

Thoughts on how to fix both cases? Should I check for a trailing / or just add one to the `--config_dir` option?

Reply:
Ok, looking more into it, I think this needs to be reversed and a `/` is added to sysconfdir when it parses the `--config_dir` option. 

I will update the PR with a new version